### PR TITLE
Interactions API

### PR DESCRIPTION
**Pull request by dlubarov**

### CODE REVIEW

File: memory/src/lib.rs [hunk 1]
```diff
@@ -116,14 +120,20 @@ where
...
+    fn global_receives(&self, machine: &M) -> Vec<Interaction<M::F>> {
+        let is_read: VirtualPairCol<M::F> = VirtualPairCol::single_main(MEM_COL_MAP.is_read);
+        let addr = VirtualPairCol::single_main(MEM_COL_MAP.addr);
+        let value = MEM_COL_MAP.value.0.map(VirtualPairCol::single_main);
+
+        let mut fields = vec![is_read, addr];
+        fields.extend(value);
+
+        let receive = Interaction {
+            fields: vec![],
```
Comment (maxgillett) [inline memory/src/lib.rs:132]: Just checking, this should be the fields vector defined above?

File: cpu/src/lib.rs [hunk 2]
```diff
@@ -60,13 +65,26 @@ where
...
+            Interaction {
+                fields,
+                count: VirtualPairCol::single_main(channel.used),
+                argument_index: machine.mem_bus(),
+            }
+        });
+        let send_general = Interaction {
+            fields: vec![],
+            count: VirtualPairCol::single_main(CPU_COL_INDICES.opcode_flags.is_bus_op),
+            argument_index: 0,
```
Comment (maxgillett) [inline cpu/src/lib.rs:85]: This should be `machine.general_bus()`, correct? And in `fields`, we would be sending `ChipChannelCols`?

**Comment by maxgillett** [comment]
Organization looks good to me.

File: memory/src/lib.rs (hunk 1 above)
Comment (dlubarov) [reply memory/src/lib.rs:132]: Oops good catch! Forgot to change IDE defaults.

File: cpu/src/lib.rs (hunk 2 above)
Comment (dlubarov) [reply cpu/src/lib.rs:85]: Yes fixed, thanks

**Pull request merged**